### PR TITLE
Query TempDB in more async way.

### PR DESCRIPTION
**Pull request by laiqu**
If database is slow for some reason, we don't want it to block whole Twisted.

### CODE REVIEW

File: python/ct/client/async_log_client_test.py [hunk 1]
```diff
@@ -113,7 +114,7 @@ def request(self, method, uri):
 
     class FakeDB(object):
         def scan_entries(self, first, last):
-            raise database.KeyError
+            raise database.KeyError("boom!")
```
Comment (pphaneuf) [inline python/ct/client/async_log_client_test.py]: Either leave the empty message, or put a better one.

File: python/ct/client/async_log_client_test.py [hunk 2]
```diff
@@ -131,13 +132,15 @@ def one_shot_client(self, json_content):
                                          test_util.DEFAULT_URI,
                                          reactor=self.clock)
 
-    def default_client(self, entries_db=None):
+    def default_client(self, entries_db=None, reactor_ = None):
```
Comment (pphaneuf) [inline python/ct/client/async_log_client_test.py]: Extra whitespace here.

File: python/ct/client/async_log_client_test.py (hunk 1 above)
Comment (laiqu) [inline python/ct/client/async_log_client_test.py]: I changed it, so we raise actual instance of Exception. Also it's nice to have boom! so we know that it's test exception (the same pattern is in monitor_test)

**Comment by pphaneuf** [comment]
LGTM

**Pull request merged**